### PR TITLE
Cfg version

### LINKED ISSUE #869: Negative config version numbers
I've just noticed that we support negative version numbers after `@version`. Is this intended? We use `strtol` for the parsing. Also, the `user_version`  field in GlobalConfig also a signed integer.

### PR DESCRIPTION
**Pull request by Kokan**
Fixes #869

Unit tests from #875

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/pragma-parser.c [hunk 1]
```diff
@@ -36,14 +36,26 @@ process_version_string(gchar *value)
...
       major = strtol(value, &end, 10);
+      if (major < 0)
+        return 0;
       if (end == p)
         {
           minor = strtol(p+1, &end, 10);
-          if (end)
+          if (minor < 0)
+            return 0;
+          if (*end == '\0' || *end == '.')
```
Comment (furiel) [inline lib/pragma-parser.c]: The `== '.'` is not needed.

**Comment by furiel** [comment]
> pragma: extract process_version_string from grammer

There is a typo in a commit message

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**